basic_bootloader/src/bootloader/process_transaction.rs

```diff
@@ -534,18 +539,11 @@ where
         let chain_id = system.get_chain_id();
 
         // Process access list
-        // Note: this operation should be performed before the hashing of the
```

Comment:
Can you pls clarify why it was removed? L1 tx can't be invalid?

Reply:
This is L2 tx, right? Not sure what you mean by the second part.
Access/authorization list parsing is done lazily. This parsing can fail if the structure is incorrect. However, after the first time you traverse it, subsequent traversals cannot fail. This fact was being used before to make tx hashing infallible (other than out of resources), which means that if you hashed a tx before doing the first parsing, it would get to an internal error. But I thought it was dangerous to keep this invariant, so I reverted to making each traversal safe (returning validation error).